datalad/support/gitrepo.py

```diff
@@ -2815,7 +2815,15 @@ def set_gitattributes(self, attrs, attrfile='.gitattributes', mode='a'):
         attrdir = op.dirname(git_attributes_file)
         if not op.exists(attrdir):
             os.makedirs(attrdir)
-        with open(git_attributes_file, mode) as f:
+
+        with open(git_attributes_file, mode + '+') as f:
+            # for append, fix existing files that do not end with \n
+            if mode == 'a':
+                f.seek(0)
+                s = f.read()
+                if len(s) > 1 and s[-1] != '\n':
```

Comment:
```suggestion
                if s and not s.endswith('\n'):
```
is a bit more pythonic to me